apps/src/templates/testPages/WebLabTest.jsx

```diff
@@ -22,14 +26,59 @@ class WebLabTest extends Component {
     this.state = {
       [STATUS_CODE_PROJECTS]: Status.WAITING,
       [STATUS_COMPUTING_IN_THE_CORE]: Status.WAITING,
+      [STATUS_BRAMBLE_MOUNTABLE]: Status.WAITING,
+      runButtonDisabled: false,
       renderCallToAction: false
     };
   }
 
-  promisifiedTestImageAccess = ({url, status}) =>
+  componentDidMount = () => {
+    window.addEventListener('message', this.updateBrambleStatus);
+  };
+
+  componentWillUnmount = () => {
+    window.removeEventListener('message', this.updateBrambleStatus);
+  };
+
+  updateBrambleStatus = (event = {}) => {
+    let message;
+    try {
+      message = JSON.parse(event.data);
+    } catch (err) {
+      console.log(err);
+    }
+
+    if (
+      event.origin === brambleConfig.studioUrl &&
+      message.type &&
+      message.type === 'bramble:readyToMount' &&
```

Comment:
nit: using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) will simplify this expression a little bit

Reply:
Forgot this was a thing now, thank you!